coderd/database/queries/licenses.sql

```diff
@@ -35,3 +35,21 @@ DELETE
 FROM licenses
 WHERE id = $1
 RETURNING id;
+
+-- name: GetManagedAgentCount :one
+-- This isn't strictly a license query, but it's related to license enforcement.
+SELECT
+	COUNT(DISTINCT wb.id) AS count
+FROM
+	workspace_builds AS wb
+JOIN
+	provisioner_jobs AS pj
+ON
+	wb.job_id = pj.id
+WHERE
+	wb.transition = 'start'::workspace_transition
+	AND wb.has_ai_task = true
+	-- Exclude failed builds since they can't use AI managed agents anyway.
+	AND pj.job_status NOT IN ('canceled'::provisioner_job_status, 'failed'::provisioner_job_status)
```

Comment:
Maybe it'll be more future-proof to whitelist instead of blacklist statuses... `pending`, `running` or `succeeded` are the only states you care about, right? This currently ignores other statuses like `canceling`, `unknown`, etc.